File: backend/src/main/java/com/crm/entity/Deal.java
package com.crm.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("deals")
public class Deal {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    @TableField("customer_id")
    private Long customerId;
    
    @TableField("opportunity_id")
    private Long opportunityId;
    
    @TableField("title")
    private String title;
    
    @TableField("amount")
    private BigDecimal amount;
    
    @TableField("contract_date")
    private LocalDate dealDate;
    
    @TableField("status")
    private String status;
    
    @TableField("remark")
    private String remark;
    
    @TableField("created_at")
    private LocalDateTime createdAt;
}

File: frontend/src/api/deal.ts
import request from './request'

export interface Deal {
  id?: number
  customerId: number
  customerName?: string
  opportunityId?: number
  title: string
  amount: number
  dealDate: string
  remark: string
  createdAt?: string
}

export const getDeals = (params?: any) => request.get('/deals', { params })
export const createDeal = (data: Deal) => request.post('/deals', data)
